drivers/hmis/app/models/hmis/reminders/reminder_generator.rb

```diff
@@ -0,0 +1,213 @@
+module Hmis
+  module Reminders
+    class ReminderGenerator
+      attr_accessor :enrollments, :project
+
+      def self.perform(...)
+        new(...).perform
+      end
+
+      # @param project [Hmis::Hud::Project]
+      # @param enrollments [ActiveRecord::Collection<Hmis::Hud::Enrollment>]
+      def initialize(project:, enrollments:)
+        self.project = project
+        self.enrollments = enrollments.preload(:client, :current_living_situations).to_a
+      end
+
+      def perform
+        results = enrollments.flat_map do |enrollment|
+          [
+            annual_assessment_reminder(enrollment),
+            aged_into_adulthood_reminder(enrollment),
+            intake_assessment_reminder(enrollment),
+            exit_assessment_reminder(enrollment),
+            current_living_situation_reminder(enrollment),
```

Comment:
reminders for `annual_assessment_reminder` and `intake_assessment_reminder` should only appear max 1 time per household. does this make them appear once per member?

Reply:
I implemented this deduplication on the front-end. I also made sorting reminders a front-end responsibility. 

Per our discussion today, when there are duplicate reminder topics, chose the reminder
* use the reminder for the currently displayed enrollment
* if none, use the reminder for the HOH
* if none, use the last reminder sorted by date

Reply:
Also, just to clarify we can still show multiple reminders for a household around current living situation. The other reminders are de-duplicated

Reply:
Yes, that's correct. Reminders about Update assessments (aging into adulthood) can also show multiple, which looks like how you've implemented it already.